contract_invoice_merge_by_partner/models/account_analytic_analysis.py

```diff
@@ -12,19 +12,18 @@ class PurchaseOrderLine(models.Model):
     def _recurring_create_invoice(self, automatic=False):
         invoice_obj = self.env['account.invoice']
         invoices = invoice_obj.browse(
-            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic)
-        )
+            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic))
         res = []
         unlink_list = []
         for partner in invoices.mapped('partner_id'):
-            inv_to_merge = invoices.filtered(lambda x: x.partner_id == partner)
-            if partner.contract_invoice_merge and (len(inv_to_merge) > 1):
-                invoices_info = inv_to_merge.do_merge()
-                res.extend(invoices_info.keys())
-                for inv_ids_list in invoices_info.values():
-                    unlink_list.extend(inv_ids_list)
+            inv_to_merge = invoices.filtered(
+                lambda x: x.partner_id.id == partner)
+            if partner.contract_invoice_merge:
+                invoices_merged = inv_to_merge.do_merge()
+                res.extend(invoices_merged)
+                unlink_list.extend(inv_to_merge)
             else:
-                res.extend(inv_to_merge.ids)
+                res.extend(inv_to_merge)
         if unlink_list:
-            invoice_obj.browse(unlink_list).unlink()
+            invoice_obj.unlink([x.id for x in unlink_list])
```

Comment:
The previous expression is correct. Yours is not, because in new API you have to make unlink over a recordset, not over a list of IDs. Even more, `unlink_list` is a list of IDs, not a recordset.